app/controllers/recipe_foods_controller.rb

```diff
@@ -0,0 +1,74 @@
+class RecipeFoodsController < ApplicationController
+  before_action :set_recipe_food, only: %i[show edit update destroy]
+
+  # GET /recipe_foods or /recipe_foods.json
+  def index
+    @recipe_foods = RecipeFood.all
+    @recipe = Recipe.find(params[:recipe_id])
+  end
+
+  # GET /recipe_foods/1 or /recipe_foods/1.json
+  def show; end
+
+  # GET /recipe_foods/new
+  def new
+    @recipe_food = RecipeFood.new
+  end
+
+  # GET /recipe_foods/1/edit
+  def edit; end
+
+  # POST /recipe_foods or /recipe_foods.json
+  # rubocop:disable
+  def create
+    @recipe = Recipe.find(params[:recipe_id])
+    @recipe_food = @recipe.recipe_foods.new(recipe_food_params)
+
+    respond_to do |format|
+      if @recipe_food.save
+        format.html { redirect_to user_recipe_path(@recipe, :user_id), notice: 'Recipe food was successfully created.' }
+        format.json { render :show, status: :created, location: @recipe_food }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+        format.json { render json: @recipe_food.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # PATCH/PUT /recipe_foods/1 or /recipe_foods/1.json
+  def update
+    respond_to do |format|
+      if @recipe_food.update(recipe_food_params)
+        format.html do
+          redirect_to user_recipe_food_url(@recipe_food), notice: 'Recipe food was successfully updated. ✅'
+        end
+        format.json { render :show, status: :ok, location: @recipe_food }
+      else
+        format.html { render :edit, status: :unprocessable_entity }
+        format.json { render json: @recipe_food.errors, status: :unprocessable_entity }
+      end
+    end
+  end
+
+  # DELETE /recipe_foods/1 or /recipe_foods/1.json
+  def destroy
+    @recipe_food.destroy
+
+    respond_to do |format|
+      format.html { redirect_to recipe_path, notice: 'Recipe food was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
```

Comment:
- [ ] I'm not able to delete a recipe because the recipe has dependencies. Please ensure the user is able to delete their own recipes.

<img width="908" alt="image" src="https://user-images.githubusercontent.com/77190203/190674744-1c6d5de8-6240-4eaa-b44d-8397331e5445.png">

Reply:
- implemented